### PR TITLE
make cluster check usable with --skip-kinds

### PR DESCRIPTION
**Pull request by clux**

### CODE REVIEW

File: shipcat_cli/src/main.rs [hunk 1]
```diff
@@ -262,6 +264,11 @@ fn build_cli() -> App<'static, 'static> {
                 .short("c")
                 .long("check")
                 .help("Check the validity of the template"))
+               .arg(Arg::with_name("skip-kinds")
+                .long("skip-kinds")
+                .takes_value(true)
+                .requires("check")
+                .help("Kinds to ignore strongest checks for for (comma separated)"))
```
Comment (gleaver) [inline shipcat_cli/src/main.rs]: 4 4

File: shipcat_cli/src/helm.rs [hunk 2]
```diff
@@ -99,7 +99,7 @@ pub async fn template(mf: &Manifest, output: Option<PathBuf>) -> Result<String>
 /// We don't validate kubernetes schemas in here, but we do validate consistency of:
 /// - labels: app.kubernetes.io/name, app.kubernetes.io/version, app.kubernetes.io/managed-by
 /// - ownerReferences (need ShipcatManifest, !controller, uid propagated, name correct)
-pub fn template_check(mf: &Manifest, reg: &Region, tpl: &str) -> Result<()> {
+pub fn template_check(mf: &Manifest, reg: &Region, skipped: &Vec<String>, tpl: &str) -> Result<()> {
```
Comment (gleaver) [inline shipcat_cli/src/helm.rs]: is there a reason this one isn't async?

File: shipcat_cli/src/helm.rs [hunk 3]
```diff
@@ -184,7 +184,7 @@ fn check_labels(mf: &Manifest, kind: &str, obj: &MetaObject) -> Result<bool> {
     // If the object doesn't get injected into the Deployment automatically
     // then it ought to have the standard version property.
     // If it changes, we should not lie about it changing (Secret + CM didn't really change)
-    if !["Secret", "ConfigMap"].contains(&kind) {
+    if !["Secret", "ConfigMap"].contains(&kind) && !skipped.contains(&kind.to_string()) {
```
Comment (gleaver) [inline shipcat_cli/src/helm.rs]: If we support `skip-kinds`, should these still be hardcoded?

File: shipcat_cli/src/cluster.rs [hunk 4]
```diff
@@ -97,19 +97,18 @@ async fn check_summary(svc: String, conf: &Config, reg: &Region) -> Result<Strin
...
 ///
 /// Helper that shells out to helm template in parallel.
-pub async fn mass_template_verify(conf: &Config, reg: &Region) -> Result<()> {
+pub async fn mass_template_verify(conf: &Config, reg: &Region, skipped: &Vec<String>) -> Result<()> {
     let svcs = shipcat_filebacked::available(conf, reg).await?;
-    assert!(conf.has_secrets());
 
     let mut buffered = stream::iter(svcs)
-        .map(move |mf| check_summary(mf.base.name, &conf, &reg))
+        .map(move |mf| check_summary(mf.base.name, &skipped, &conf, &reg))
```
Comment (gleaver) [inline shipcat_cli/src/cluster.rs]: doesn't this need `await`ing?

File: shipcat_cli/src/helm.rs (hunk 2 above)
Comment (clux) [reply shipcat_cli/src/helm.rs]: yep, is has all data and it's not doing IO

File: shipcat_cli/src/cluster.rs (hunk 4 above)
Comment (clux) [reply shipcat_cli/src/cluster.rs]: No :-)

Comment (clux) [reply shipcat_cli/src/cluster.rs]: No :-)

Missing await is only a problem if we explicitly trashed the output var  (like by doing `let _ = fncall()`)

File: shipcat_cli/src/helm.rs (hunk 3 above)
Comment (clux) [reply shipcat_cli/src/helm.rs]: Probably. It's kind of assumed that deployments change when secrets/configs change.

**Pull request merged**